### PR TITLE
revert auth changes

### PR DESCRIPTION
**Pull request by Will-ShaoHua**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description

1. reauthConnection changes makes `ToolkitConnectionManager` not switching to the connection after auth succeed
2. using `ApplicationManager.getApplication().executeOnPooledThread` will make the auth process executed in a non-blocking way in the background which makes it hard to run logic when login succeed or fail. 

## Checklist
- [ ] My code follows the code style of this project
- [ ] I have added tests to cover my changes
- [ ] A short description of the change has been added to the **[CHANGELOG](https://github.com/aws/aws-toolkit-jetbrains/blob/master/CONTRIBUTING.md#contributing-via-pull-requests)** if the change is customer-facing in the IDE.
- [ ] I have added metrics for my changes (if required)
 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt [hunk 1]
```diff
@@ -180,8 +177,13 @@ fun loginSso(project: Project?, startUrl: String, region: String, requestedScope
 }
 
 @VisibleForTesting
-internal fun reauthConnection(project: Project?, connection: ToolkitConnection): BearerTokenProvider =
-    reauthConnectionIfNeeded(project, connection)
```
Comment (Will-ShaoHua) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt:184]: https://github.com/aws/aws-toolkit-jetbrains/commit/317eb696b0028ff6066204d4f0f7b47252b3c4e0#diff-a8660e33b45cfd1cb158ffa3a56fb182b9f0c5af9ae1a26e6d7fc049bbf99e02R183-R184

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt [hunk 2]
```diff
@@ -219,19 +221,15 @@ fun AwsBearerTokenConnection.lazyIsUnauthedBearerConnection(): Boolean {
...
-            } catch (e: ProcessCanceledException) {
-                getLogger<ToolkitAuthManager>().debug { "User canceled the login flow" }
-            }
+    return reauthProviderIfNeeded(project, tokenProvider)
+}
+
+fun reauthProviderIfNeeded(project: Project?, tokenProvider: BearerTokenProvider): BearerTokenProvider {
+    maybeReauthProviderIfNeeded(project, tokenProvider) {
+        runUnderProgressIfNeeded(project, message("credentials.pending.title"), true) {
+            tokenProvider.reauthenticate()
```
Comment (Will-ShaoHua) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt:230]: https://github.com/aws/aws-toolkit-jetbrains/commit/317eb696b0028ff6066204d4f0f7b47252b3c4e0#diff-a8660e33b45cfd1cb158ffa3a56fb182b9f0c5af9ae1a26e6d7fc049bbf99e02L242-L248

**Review by rli — APPROVED** [review]

File: plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt [hunk 3]
```diff
@@ -219,19 +221,15 @@ fun AwsBearerTokenConnection.lazyIsUnauthedBearerConnection(): Boolean {
 
 fun reauthConnectionIfNeeded(project: Project?, connection: ToolkitConnection): BearerTokenProvider {
     val tokenProvider = (connection.getConnectionSettings() as TokenConnectionSettings).tokenProvider.delegate as BearerTokenProvider
-    val needReAuth = maybeReauthProviderIfNeeded(project, tokenProvider) {
-        ApplicationManager.getApplication().executeOnPooledThread {
```
Comment (Will-ShaoHua) [inline plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt:223]: with current implementation, by using `executeOnPoolThread`, we have no way to run logic onAuthSuccess or onAuthFailure. Therefore revert this change.

Comment (Will-ShaoHua) [reply plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/core/credentials/ToolkitAuthManager.kt:223]: the original intention of this change is to hide the progress indicator while users are going through the browser auth flow
